src/evict.c

```diff
@@ -697,7 +706,7 @@ int performEvictions(void) {
                  * across the dbAsyncDelete() call, while the thread can
                  * release the memory all the time. */
                 if (server.lazyfree_lazy_eviction) {
-                    if (getMaxmemoryState(NULL, NULL, NULL, NULL) == C_OK) {
+                    if (getMaxmemoryState(NULL, NULL, NULL, NULL, server.key_eviction_memory) == C_OK) {
```

Comment:
should we update `mem_used` too here?
```suggestion
                    if (getMaxmemoryState(NULL, &mem_used, NULL, NULL, server.key_eviction_memory) == C_OK) {
```

Reply:
I think we should not change it, otherwise some existing test cases are broken

Reply:
I see - updating `mem_used` here essentially moves the goal post. I think the variable name (`mem_used`) is the source of my confusion. agreed with you that we don't need to update `mem_used`